app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,105 @@
+class OrderItemsController < ApplicationController
+
+
+  def index
+    @order_items = OrderItem.joins(:order).where('orders.status' => 'pending').where('orders.id' => session[:order_id])
+  end
+
+  def new
+    @item = OrderItem.new
+    @product = Product.find(params[:product_id])
+    @merchant = Product.find(params[:product_id]).user_id
```

Comment:
This `find` method call will call the database again to retrieve the product in order to return the `user_id`. You want to avoid calling the database unnecessarily so this should be updated to use the existing variable that you have which contains the Product's information.